### ISSUE TITLE
Support hiding of certain files from build output

### ISSUE THREAD

**Issue by agoose77** [jupyter-book/mystmd#1666]
This might end up being a theme requirement.

Some users would like to be able to embed content from a source document without making that document visible in the built target (site). Sphinx has `orphan` annotations in the frontmatter.

See https://github.com/orgs/jupyter-book/discussions/1663#discussioncomment-11331633 for more details.

**Comment by choldgraf** [comment]
@agoose77 I cleaned up the issue body a bit so that it was clearer what kind of end-functionality we needed from a user's perspective

**Comment by fperez** [comment]
Yes, I had several of these in my old site that I'm for now putting in the myst toc, but would very much prefer to consider "internal implementation". 

For now I'm putting them at the very end of the toc, but happy to see this being tracked. Thanks!

**Comment by stefanv** [comment]
Another documentation system I use flag pages as `hidden: true`.

**Comment by bsipocz** [comment]
I finally found this issue after running into problems with orphan pages, and can confirm that using `hidden` works around the problem.

https://mystmd.org/guide/table-of-contents#hidden-in-toc

I would say that is a workaround though, but if you consider it good enough then maybe we can close this issue.

(I'll be opening a separate one as the `--strict` build option should complain about the orphan pages as their linking will not work properly without listing them in the TOC)
